source/common/http/conn_manager_impl.cc

```diff
@@ -1531,18 +1531,30 @@ void ConnectionManagerImpl::ActiveStream::onBelowWriteBufferLowWatermark() {
 }
 
 Tracing::OperationName ConnectionManagerImpl::ActiveStream::operationName() const {
+  ASSERT(
```

Comment:
Argh, github lost my comment. The consumer of https://github.com/envoyproxy/envoy/blob/master/source/common/tracing/http_tracer_impl.cc#L283 needs a nullptr check.

Reply:
This function and the ones below should be changed to return an optional, otherwise it's still going to crash. Then the caller can be fixed to ignore, etc. depending on what it's doing.

Reply:
@mattklein123 These methods are there to satisfy the Tracing::Config interface. Do you want me to change all of its method signatures? I'm not following...

Reply:
Yes, you need to change the signatures of the functions. The other option is to always have a "null tracing config" so there is always something to pull from (kind of like the NullSpan) but I'm not sure this makes sense.